torchio/transforms/augmentation/composition.py

```diff
@@ -41,12 +41,15 @@ def is_invertible(self):
         return all(t.is_invertible() for t in self.transforms)
 
     def inverse(self):
-        transforms = [
-            transform.inverse()
-            for transform in self.transforms
-            if transform.is_invertible()
-        ]
-        return Compose(reversed(transforms))
+        transforms = []
+        for transform in self.transforms:
+            if transform.is_invertible():
+                transforms.append(transform.inverse())
+            else:
+                message = f'Skipping {transform.name} as it is not invertible'
+                raise UserWarning(message)
+        transforms.reverse()
+        return Compose(transforms)
```

Comment:
so you can still do the inverse although the `is_invertible()` methode will return false ?
then when `is_invertible` method will be used ?

Reply:
I was discussing this with @pedrob37 yesterday. If you apply a composition of e.g. affine and blur and infer a segmentation, you'd like to invert the affine only because 1) what you want is coming back to the original space and 2) it's not possible to undo the blur anyway. I think it's still nice to be able to call this method, getting a warning.

Reply:
Yes I agree,